provider/provider.go

```diff
@@ -164,24 +167,26 @@ func (clientTLS *ClientTLS) CreateTLSConfig() (*tls.Config, error) {
 	cert := tls.Certificate{}
 	_, errKeyIsFile := os.Stat(clientTLS.Key)
 
-	if _, errCertIsFile := os.Stat(clientTLS.Cert); errCertIsFile == nil {
-		if errKeyIsFile == nil {
-			cert, err = tls.LoadX509KeyPair(clientTLS.Cert, clientTLS.Key)
-			if err != nil {
-				return nil, fmt.Errorf("Failed to load TLS keypair: %v", err)
+	if len(clientTLS.Cert) > 0 && len(clientTLS.Key) > 0 {
```

Comment:
This is to wrap the certificate loading only if it is indeed configured.  Otherwise things fail when you only use `clientTLS.InsecureSkipVerify=true`.  This was causing issues when trying to connect to Mesos master that has a self-signed cert.   Do you want to move that to a separate PR?